client/java/src/main/java/io/openlineage/client/transports/Transport.java

```diff
@@ -23,4 +23,12 @@ enum Type {
   Transport(@NonNull final Type type) {}
 
   public abstract void emit(OpenLineage.RunEvent runEvent);
+
+  public void emit(OpenLineage.DatasetEvent datasetEvent) {
```

Comment:
Why not `abstract`? Do not want to enforce implementation in Transports made for previous versions?

Reply:
Users may have custom transport implementations. Making it `abstract` would enforce implementation on their side and would break their code even though they may have no interest in static metadata.

Reply:
What would be the usage of `emit(String)`? Our clients define models that make it clear the types allowed and enforces a structure. Allowing for `string` type allows for flexibility, but introduces the likelihood of invalid events being emitted.